### PR TITLE
feat: /ref support, suggestions, better references, debug and warnings

### LINKED ISSUE #11: Auto-fix redirects, disallowed extensions and missing anchors
It would be cool if we could run the tool with `--fix` in order to replace redirected links automatically.

### PR DESCRIPTION
**Pull request by dcdunkan**
changes in this pq:

- local links starting with '/' were being ignored before. that had to be fixed as now we have /ref links and other local links referenced from /ref pages.
- jaro-winkler impl. for suggesting possible fixes for missing anchors
- fixed more of wrong references. these are needed for finding the line and column
- force old api ref links to new /ref pages (local alts available)
- `--fix` argument for auto-fixing stuff like redirections, extensions and missing anchors
- debug mode
- inaccessible websites: ignores cloudflare protected sites now

Closes #11

### CODE REVIEW

**Comment by KnorpelSenf** [comment]
Please ping me when ready

**Comment by dcdunkan** [comment]
@KnorpelSenf I believe it's ready although it lacks few more testing.

**Comment by KnorpelSenf** [comment]
Very cool, I'll do some of that testing right now

File: README.md [hunk 1]
```diff
@@ -1,7 +1,27 @@
...
+
+Link checker currently covers the following types of issues:
+
+- Redirections
+- Empty responses
+- Empty DOM contents
+- Empty anchors
+- Missing anchors
+- Non-OK responses
+- Wrong extensions
```
Comment (KnorpelSenf) [inline README.md]:
```suggestion
- Wrong file extensions on internal links
```

File: README.md [hunk 2]
```diff
@@ -51,3 +96,10 @@ Supported arguments:
 
 - `--module`: Relative path or absolute local or remote URL to the module entry point to check. For example,
   https://deno.land/x/grammy/mod.ts.
+
+## Github workflow
+
+This repository also contains a Github workflow for checking for issues and reporting them in the website's Github repository issue section.
+
+TODO: Explain how to configure the workflow.
```
Comment (KnorpelSenf) [inline README.md:104]: Do you want to do this before merging?

File: constants.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    "Cache-Control": "no-cache",
+  },
+  mode: "cors",
+} as const;
+
+export const DEFAULT_GITHUB_API_ROOT = "https://api.github.com";
+
+export const SEARCH_PANIC_MESSAGE = `\
+====================================================================================
+PANIK. This shouldn't be happening. The search strings are supposed to have at least
```
Comment (KnorpelSenf) [inline constants.ts]:
```suggestion
PANIC. This shouldn't be happening. The search strings are supposed to have at least
```

**Review by KnorpelSenf — APPROVED** [review]
Rubber-stamp LGTM. I did not review this very thoroughly, but I also don't want to block this any further. If issues arise in the future, I'll report them then.

File: README.md (hunk 2 above)
Comment (dcdunkan) [reply README.md:104]: No. I actually forgot the flow of doing this. I'll do it in another commit after I verify the steps, and link it to the original repository.

**Pull request merged**